### DISCUSSION TITLE
Actual site URL in the auth callback URL

### DISCUSSION THREAD

**Discussion by ilyabo** [supabase/supabase#3428] [Questions] [answered]
Is it possible to use the actual site URL in the auth callback URL for external OAuth providers? Maybe at least with the Pro plan?

It doesn't look very convincing or trustworthy with the randomly generated hostname:

<img src=https://user-images.githubusercontent.com/351828/136155124-b53dfc19-9f91-4281-b552-e364e129b175.png width=300>

I tried adding a URL-rewrite on Vercel (from `https://…supabase.co/auth/v1/callback` to `https://mysiteurl.com/auth/v1/callback`), but I am getting this error when trying to authenticate with GitHub:
```
?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.
```

**Comment by awalias** [comment, marked as answer]
you have the right approach, however we need to add the ability for you to overwrite the url passed to the auth API server (gotrue) on our side, maybe we can add this as an advanced settings option for auth, will make an issue internally, thanks for flagging

**Reply by ilyabo** [reply]
That's cool, thank you!

**Reply by awalias** [reply]
https://github.com/supabase/gotrue/issues/233

**Reply by laygir** [reply]
this really landed as a monthly paid add-on instead of a new feature for already paying pro plan customers, right?
